src/TemplateContainer.tsx

```diff
@@ -1,52 +1,53 @@
 import * as React from 'react';
+import type { DragHandleProps, RenderTemplate } from '.';
 
 interface Props<I, C, T> {
   item: I;
-  template: new (props: any, context?: any) => T;
+  renderTemplate: RenderTemplate<I, C, T>;
   itemSelected: number;
   anySelected: number;
-  dragHandleProps: object;
-  commonProps: C;
+  dragHandleProps: DragHandleProps;
+  commonProps?: C;
```

Comment:
Let's drop `commonProps` entirely. It's not necessary when using the render-props pattern.

Reply:
[054e748](https://github.com/StreakYC/react-draggable-list/pull/62/commits/054e7481c01d1a2febdd4cb81b4ce9e1b959daac)